app/src/main/java/com/pb/nex26dv/shippingapp/orderplacing.java

```diff
@@ -0,0 +1,566 @@
+            width.setEnabled(false);
+        }
+
+        @Override
+        protected Void doInBackground(String... strings) {
+            String fileUrl = strings[0];   // -> http://10.50.14.48/fedex/187143416/outbound/label/f9102b14a5c74c3f8d57ed41568d45d0.pdf
+            String fileName = strings[1];  // -> label.pdf
+            String extStorageDirectory = Environment.getExternalStorageDirectory().toString();
+            File folder = new File(extStorageDirectory, "/label");
+            folder.mkdir();
+
+            File pdfFile = new File(folder, fileName);
+
+            try {
+                pdfFile.createNewFile();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            FileDownloader.downloadFile(fileUrl, pdfFile);
+            return null;
+        }
+
+        @Override
+        protected void onPostExecute(Void aVoid) {
+            super.onPostExecute(aVoid);
+            Toast.makeText(orderplacing.this, "Label downloaded successfully", Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    private String makeDimPostCall(String requestData) {
+        HttpURLConnection httpcon;
+        String url = "http://10.220.66.60:5000/getDim";
+        String data = null;
+        String result = null;
+        try {
+            //Connect
+            httpcon = (HttpURLConnection) ((new URL(url).openConnection()));
+            httpcon.setDoOutput(true);
+            httpcon.setRequestProperty("Content-Type", "application/json");
+            //httpcon.setRequestProperty("Accept", "application/json");
+            httpcon.setRequestMethod("POST");
+            httpcon.connect();
+
+            //Write
+            OutputStream os = httpcon.getOutputStream();
+            BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(os, "UTF-8"));
+            writer.write(requestData);
+            writer.close();
+            os.close();
+
+            //Read
+            BufferedReader br = new BufferedReader(new InputStreamReader(httpcon.getInputStream(), "UTF-8"));
+
+            String line = null;
+            StringBuilder sb = new StringBuilder();
+
+            while ((line = br.readLine()) != null) {
+                sb.append(line);
+            }
+
+            br.close();
+            result = sb.toString();
+
+        } catch (UnsupportedEncodingException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return result;
+    }
+
+
+    public List<Float> getFinalDimList() {
+        List<Float> dim = new ArrayList<>();
+        if (secondResponse != null) {
+
+            l = Float.parseFloat(firstResponse.substring(0, firstResponse.indexOf(',')));
+            b = Float.parseFloat(firstResponse.substring(firstResponse.indexOf(',') + 1, firstResponse.length() - 1));
+
+            h = Float.parseFloat(secondResponse.substring(0, secondResponse.indexOf(',')));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

If the indexOf() search on line 533 does not find the substring and returns -1, the call to substring() on line 533 will throw IndexOutOfBoundsException.